### ISSUE TITLE
Environment configs

### ISSUE THREAD

**Issue by confuser** [apigee-127/magic#17]
Trying to use environment specific configurations on Edge. However, it only seems to pick up default.yaml on a `test` environment, and never `test.yaml`.

I've logged the environment variables, and `APIGEE_ENVIRONMENT` contains the correct value. However, magic is looking for `A127_ENV` or `NODE_ENV` when attempting to load the environment config, which don't appear to be defined.

Is it recommended to define either `NODE_ENV` or `A127_ENV` manually based on the `APIGEE_ENVIRONMENT` at the application level before calling `init` instead?

**Comment by theganyo** [comment]
Yes, to link the two together you could simply put something like this at the top of your app file:

`process.env.NODE_ENV=process.env.APIGEE_ENVIRONMENT`

**Comment by confuser** [comment]
Thanks @theganyo what's the reason for not using `APIGEE_ENVIRONMENT` instead within this module to specify the configuration file?

**Comment by theganyo** [comment]
There was no particular intent to it other than NODE_ENV being the standard method of control for this kind of thing in node. Retrospectively, I guess one could argue that there may be times that users don't want to use APIGEE_ENVIRONMENT for this purpose, but I'm not sure how strong an argument that really is.

**Comment by confuser** [comment]
Thanks for clarifying, I only ask as coming from a native node.js background, it seems odd to me to specify the NODE_ENV at the application/code layer rather than having the process manager specify it.

Can this stance be documented somewhere if it isn't already? My colleagues have also ran into this issue before.

**Comment by theganyo** [comment]
I think that's a fair point, it would be preferable that Edge set NODE_ENV as node developers would expect.

**Comment by confuser** [comment]
That would be great, especially as many modules in the ecosystem such as express have slight behavioural differences when run under `NODE_ENV=production`
